packages/material-ui-lab/src/Alert/Alert.d.ts

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { StandardProps } from '@material-ui/core';
+import { PaperProps } from '@material-ui/core/Paper';
+
+export type Color = 'success' | 'info' | 'warning' | 'error';
+
+export interface AlertProps extends StandardProps<PaperProps, AlertClassKey, 'variant'> {
+  /**
+   * The action to display.
+   */
+  action?: React.ReactNode;
+  /**
+   * Override the default text for the *close popup* icon button.
+   *
+   * For localization purposes, you can use the provided [translations](/guides/localization/).
+   */
+  closeText?: string;
+  /**
+   * Main color for the Alert, picked from theme palette.
+   */
+  color?: Color;
```

Comment:
This is my only major gripe.  I really think we should consider naming this `type`.  For one reason, because we don't actually know what the "color" is that the user has actually set.  It's perfectly possible that two alerts could have the same color.  Furthermore... `export type Color = 'success' | 'info' | 'warning' | 'error';` is in itself a code smell since those things listed are not themselves colors.

For another reason, of the 18 benchmark examples:
- 7 of them use `type`
- 4 of them use `variant` (note: we are already using `variant` for another purpose)
- 2 of them use `intent`
- 1 each use `role`, `color`, `severity`, `kind`, `status`

Reply:
ok.. why not both then?  I guess I was moreso trying to point out that this is controlling more than just the color (i.e. the icon) and you run into problematic questions like:

> what icon should be used for `color="secondary"`

whereas if you keep them separate you don't have this problem.  more than that.. I'm not convinced `primary` and `secondary` actually even make sense in the content of an alert's _type_.  Color.. ok sure.  but the type?  What's a `primary` alert? an error?

If there is an issue for dynamic prop support from the theme please link it here.

Reply:
I don't have a strong point of view on this matter. I have personally built a custom component to solve this problem in the past and called the prop: `variant` and like the simplicity of a single prop even if it mean losing some of the subtilities.

However, the mixed approach: type (that determines the icon and the default color) + color (that can override the color from type) + icon (that can override the icon from type) + iconMapping (that can extend the scope of the type supported) props could work.

@mbrookes What do you think?